src/librustdoc/clean/inline.rs

```diff
@@ -303,7 +302,7 @@ fn merge_attrs(
         both.extend_from_slice(old_attrs);
         (
             if let Some(new_id) = parent_module {
-                Attributes::from_ast(old_attrs, Some((inner, new_id)))
+                Attributes::from_ast(old_attrs, Some((inner, new_id.into())))
```

Comment:
from_ast should also take a real id, this is the only place it's called with Some.